osf/management/commands/migrate_deleted_date.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import connection, transaction
+from framework.celery_tasks import app as celery_app
+
+logger = logging.getLogger(__name__)
+
+TABLES_TO_POPULATE_WITH_MODIFIED = [
+    'osf_comment',
+    'osf_institution'
+]
+
+POPULATE_COLUMNS = [
+    'SET statement_timeout = 10000; UPDATE osf_basefilenode SET deleted = deleted_on WHERE id IN (SELECT id FROM osf_basefilenode WHERE deleted_on IS NOT NULL AND deleted IS NULL LIMIT %s) RETURNING id;',
+    'SET statement_timeout = 10000; UPDATE osf_abstractnode SET deleted= CASE WHEN deleted_date IS NOT NULL THEN deleted_date ELSE last_logged END WHERE id IN (SELECT id FROM osf_abstractnode WHERE is_deleted AND deleted IS NULL LIMIT %s) RETURNING id;',
+    'SET statement_timeout = 10000; UPDATE osf_privatelink PL set deleted = NL.date from osf_nodelog NL, osf_privatelink_nodes pl_n WHERE NL.node_id=pl_n.abstractnode_id AND pl_n.privatelink_id = pl.id and PL.id in (SELECT id FROM osf_privatelink WHERE is_deleted AND deleted IS NULL LIMIT %s) RETURNING PL.id;',
```

Comment:
Your private_link query doesn't look right to me.  PrivateLinks can belong to many nodes, and nodes can have lots of logs.   This is selecting one of the dates from the join of a particular private link to all of its nodes and all of its logs - the one I spot checked had a deleted_date that matched an osfstorage file log.

It looks like when a private link is deleted we add a VIEW_ONLY_LINK_REMOVED log to all of its nodes, but that log doesn't have any identifying information about the link in the log itself.

Given that, would the PrivateLink.modified date be appropriate here?

Reply:
I didn't consider that case. I agree the modified date would be more accurate.